app/models/foreman_tasks/task/dynflow_task.rb

```diff
@@ -65,6 +65,14 @@ def execution_plan(silence_exception = true)
       nil
     end
 
+    def frozen
+      delayed_plan.try(:frozen)
+    end
+
+    def delayed_plan
+      ForemanTasks.dynflow.world.persistence.load_delayed_plan(external_id)
```

Comment:
How about returning nil early if the task is not in scheduled state so we won't hit the db 20 times each time a user opens the task index just to see that there are no delayed plans for the tasks?